### ISSUE TITLE
Support $EDITOR with arguments

### ISSUE THREAD

**Issue by vgrigoriu** [PawelLipski/git-machete#619]
# Overview

If I set `EDITOR` to something like `code --wait --new-window`, `git machete` will not recognize it.

## Expected behavior/command output

Parse the value and look for the first item as the executable on the path, not the whole string.

## Actual behavior/command output

```
get_default_editor(): '$EDITOR' (code --wait --new-window) is not available
```

## `git` and `git machete` versions

```
git --version
git version 2.32.1 (Apple Git-133)
git machete --version
git machete version 3.11.3
```

**Comment by PawelLipski** [comment]
Good spot, `git` itself indeed parses `$EDITOR`:

```
GIT_EDITOR="vim -p" git commit --allow-empty
```

opens `vim -p .../.git/COMMIT_EDITMSG`.

Should be a quick fix, I'll include that in the subsequent release.

**Comment by vgrigoriu** [comment]
Thanks for the quick response, and thanks for an awesome application, I use it daily!

**Comment by PawelLipski** [comment]
Fix released in v3.11.6. Already available in PyPI and Homebrew tap, should be available via other channels soon.

Btw, how did you learn of this project? Was it via [Ben Congdon's post](https://benjamincongdon.me/blog/2022/07/17/In-Praise-of-Stacked-PRs/)?

**Comment by vgrigoriu** [comment]
Yes, via that post.
